### DISCUSSION TITLE
Listening on a specific interface (or IP address)

### DISCUSSION THREAD

**Discussion by adam-devel** [dani-garcia/vaultwarden#5292] [Q&A] [answered]
How does one get vaultwarden to listen on a chosen network interface? or more precisely, listen on a specific IP address.

Network interfaces are assigned distinct IP addresses. Programs listen through an interface by binding a socket to the IP address associated with the interface.

Vaultwarden has no configuration option for choosing the IP address it binds to. instead it expects a Domain name. how does that work ? vaultwarden has to bind to an IP address at some point. the TCP protocol has no notion of domains.

Does vaultwarden resolve the domain through DNS at startup then bind to the resolved IP address? if so, would it be sufficient to add a new domain name in `/etc/hosts` and associate it with the IP address of the desired interface and then use said domain in the config ? would that make bailtwarden listen on the chosen interface ?

**Comment by stefan0xC** [comment, marked as answer]
Vaultwarden uses the [Rocket framework](https://rocket.rs/). In the container we set the address to `0.0.0.0` and the port to `80`:
https://github.com/dani-garcia/vaultwarden/blob/9cd400db6c5da858a4f49eb883469cbd6cb7337d/docker/Dockerfile.debian#L171-L172

If you run it locally it will bind to the Rocket [defaults](https://rocket.rs/guide/v0.5/configuration/#overview) instead.
```[2024-12-14 06:41:18.045][start][INFO] Rocket has launched from http://127.0.0.1:8000```
